### PR TITLE
Bug/allocator always pub

### LINKED ISSUE #48: Should allocator keep publishing if the messages doesn't change?
If the allocator receives multiple identical `rov_forces` messages in a row, it currently only publishes `thruster_forces` only for the first of the identical messages.

See
https://github.com/vortexntnu/rov-control/blob/32ac0de16fb7b2aa430bf051ce8a843192619a7d/vortex_allocator/src/allocator_ros.cpp#L81

I don't think this is the desired behaviour. It was probably done at some point for performance reasons, but I suggest removing this feature so that one message in always causes one message out.

Actually I think the reason was that out manipulator code was very CPU heavy, and we usually used the manipulators only when not using the thrusters, so this feature would avoid taking resources away from the manipulators.

### PR DESCRIPTION
**Pull request by mortenfyhn**
Fixes #48 and cleans up the allocator a bit. A lot of renames, using a unique pointer so we don't have to worry about `delete`, and some more stuff.

### CODE REVIEW

**Comment by mortenfyhn** [comment]
Never mind the tests...

File: vortex_allocator/include/vortex_allocator/allocator_ros.h [hunk 1]
```diff
@@ -12,25 +12,24 @@
 class Allocator
 {
 public:
-  explicit Allocator(ros::NodeHandle nh);
-  void callback(const geometry_msgs::Wrench &msg);
+  Allocator(ros::NodeHandle nh);
```
Comment (sindrehan) [inline vortex_allocator/include/vortex_allocator/allocator_ros.h]: Roslint:
`vortex_allocator/include/vortex_allocator/allocator_ros.h:15:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]`

File: vortex_allocator/src/allocator_ros.cpp [hunk 2]
```diff
@@ -8,16 +9,20 @@
 #include "vortex/eigen_typedefs.h"
 #include "vortex/eigen_helper.h"
 
-Allocator::Allocator(ros::NodeHandle nh) : nh(nh)
+Allocator::Allocator(ros::NodeHandle nh)
+ :
```
Comment (sindrehan) [inline vortex_allocator/src/allocator_ros.cpp]: Roslint: 
`vortex_allocator/src/allocator_ros.cpp:13:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]`

**Review by sindrehan — CHANGES_REQUESTED** [review]
Roslint is whining, see comments.

**Comment by sindrehan** [comment]
@mortenfyhn Why ignore the tests?

**Review by sindrehan — APPROVED** [review]

**Pull request merged**